postgres/tests/test_replication_slot.py

```diff
@@ -14,14 +14,29 @@
 def test_physical_replication_slots(aggregator, integration_check, pg_instance):
     check = integration_check(pg_instance)
     redo_lsn_age = 0
-    with psycopg2.connect(host=HOST, dbname=DB_NAME, user="postgres", password="datad0g") as conn:
+
+    conn = psycopg.connect(host=HOST, dbname=DB_NAME, user="postgres", password="datad0g", autocommit=True)
```

Comment:
As a follow up to this PR, we can remove the `slot_persistence` tag on this metric, since we can only view temporary slots within the context of the same session

https://pgpedia.info/p/pg_create_physical_replication_slot.html

> If the temporary parameter is provided and set to TRUE, the replication slot will not be stored permanently and will only be available for the duration of the current session.

that would mean this test could use the `psycopg` context manager to handle closing the connection & can get rid of this phys_3 slot creation / validation